src/runtime_src/tools/xclbinutil/XclBinClass.cxx

```diff
@@ -1234,6 +1229,27 @@ XclBin::appendSections(ParameterSectionData& _PSD)
     pSection->purgeBuffers();
     pSection->readJSONSectionImage(ptPayload);
 
+    if (eKind == PARTITION_METADATA) {
```

Comment:
This functional should not be done in the append section.  Instead, this functionality should be done in XclBInUtilMain::main_() right before "dumping the sections" (at around line 536).

Reply:
addressed.